src/librustc_passes/ast_validation.rs

```diff
@@ -430,6 +439,13 @@ fn validate_generics_order<'a>(
         }
         err.emit();
     }
+
+    // FIXME(const_generics): we shouldn't have to abort here at all, but we currently get ICEs
+    // if we don't. Const parameters and type parameters can currently conflict if they
+    // are out-of-order.
+    if is_out_of_order && found_type && found_const {
```

Comment:
Nit: can use `!out_of_order.is_empty()` directly here (it's not necessary to consume `out_of_order` in the loop above).